### PR TITLE
docs: add info about openssl, pkg-config deps

### PR DESCRIPTION
**Pull request by shantanoo-desai**
- Without previous installation of `openssl-dev`, `pkg-config`,
  `cargo install cargo-generate` will fail.
- Potential solution is to install the respective dependencies beforehand

Tested On: Windows10 Pro + WSL2 (Ubuntu-20.04 LTS) + rustc > v1.51
Possible Solution found on: https://github.com/sfackler/rust-openssl/issues/951

Signed-off-by: Shan Desai <shantanoo.desai@gmail.com>

**What this PR does / why we need it**:

Additional Information in Documentation for creating a custom discovery handler.

**Special notes for your reviewer**:

**If applicable**:
- [x] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] added code adheres to standard Rust formatting (`cargo fmt`)
- [ ] code builds properly (`cargo build`)
- [ ] code is free of common mistakes (`cargo clippy`)
- [ ] all Akri tests succeed (`cargo test`)
- [ ] inline documentation builds (`cargo doc`)
- [ ] version has been updated appropriately (`./version.sh`)

### CODE REVIEW

**Comment by kate-goldenring** [comment]
Hi Shan! Thanks for finding this. Would it make sense to use the `--features vendored-openssl` to avoid having to install openssl? [`vendored-openssl`](https://github.com/cargo-generate/cargo-generate#using-cargo-with-vendored-openssl) does require other packages, though, so its a trade off. What do you think?

**Comment by shantanoo-desai** [comment]
@kate-goldenring fair point. From a trade-off perspective `cargo install cargo-generate` is extremely resource intensive (my observation) and not sure if `--features vendored-openssl` actually makes it easier or not. 

Better approach would be to give the reader the choice, either dependencies installation or enable the `features` flag.

**Comment by kate-goldenring** [comment]
@shantanoo-desai, is there a way to point out to their documentation so that they have the option, and if their dependencies change we don't need to update our docs?

**Comment by shantanoo-desai** [comment]
I was trying to make some sense of the Akri MQTT integration and it turns out The Eclipse Paho Group which is responsible for providing MQTT clients in all programming languages do use the `--features` flag 
Since they essentially are using a rust wrapper on the C library which consists of SSL bindings.

[paho.mqtt.rust Linking OpenSSL Statically](https://github.com/eclipse/paho.mqtt.rust#linking-openssl-statically)

**Comment by kate-goldenring** [comment]
> I am not sure why the rust opc-ua client doesn't have such bindings for SSL Certificates.

Which rust opc ua client are you referring to? Akri's OPC UA discovery handler or a specific crate?

For this PR, i think it would be best to do remove `cargo install cargo-generate` and replace it with a line that links to their install instructions, such as: "... install [cargo-generate](https://github.com/cargo-generate/cargo-generate#installation) ..." 

How does that sound?

**Comment by shantanoo-desai** [comment]
> Which rust opc ua client are you referring to? Akri's OPC UA discovery handler or a specific crate?

The [`opcua-client` crate](https://github.com/locka99/opcua) used in Akri's OPC-UA Discovery Handler Implementation.

So for this PR it should look something like the following:

```diff
-> NOTE: `cargo-generate` depends on `openssl-dev` and `pkg-config` and need to be installed beforehand
-
-__GNU/Linux__:
-
-```sh
-apt install -y openssl-dev pkg-config
-```
-
-__macOS__:
+Install [`cargo-generate`](https://github.com/cargo-generate/cargo-generate#installation) and specify the name of the project with the `--name` parameter.
```

**Comment by kate-goldenring** [comment]
> The opcua-client crate used in Akri's OPC-UA Discovery Handler Implementation.

Akri's OPC UA discovery handler is simply querying opc ua server's discovery endpoints (which do not require authentication). The client is [not creating credentials for itself](https://github.com/deislabs/akri/blob/main/discovery-handlers/opcua/src/wrappers.rs#L48) which may be why no bindings for ssl certs are required.

File: docs/discovery-handler-development.md [hunk 1]
```diff
@@ -148,24 +148,10 @@ deleted Configuration.
...
-> NOTE: `cargo-generate` depends on `openssl-dev` and `pkg-config` and need to be installed beforehand
-
-__GNU/Linux__:
-
-```sh
-apt install -y openssl-dev pkg-config
-```
-
-__macOS__:
+Install by following the [`cargo-generate` instruction](https://github.com/cargo-generate/cargo-generate#installation) and
```
Comment (kate-goldenring) [inline docs/discovery-handler-development.md:151]: I'd reword to clarify the two different steps: installing the tool and pulling down the template using the tool. Something like: Install [`cargo-generate`](https://github.com/cargo-generate/cargo-generate#installation) and use the tool to pull down Akri's template, specifying the name of the project with the `--name` parameter.

Comment (shantanoo-desai) [reply docs/discovery-handler-development.md]: resolved

**Review by kate-goldenring — APPROVED** [review]

**Comment by kate-goldenring** [comment]
@shantanoo-desai can you add your mqtt proposal from commit https://github.com/deislabs/akri/pull/339/commits/00d03933a501d085a901b4b9b69455a5f58d9876 to a different PR?

File: docs/extensibility.md [hunk 2]
```diff
@@ -31,8 +31,11 @@ the Agent, which hosts the `Registration` service defined in the gRPC interface.
 ### Use `cargo generate` to clone the Discovery Handler template
 Pull down the [Discovery Handler template](https://github.com/kate-goldenring/akri-discovery-handler-template) using
```
Comment (kate-goldenring) [inline docs/extensibility.md:32]: these two sentences are redundant. Can you delete the first.

Comment (shantanoo-desai) [reply docs/extensibility.md:32]: Do you mean the h3 heading or the sentence starting with _Pull down ..._?
